src/test/java/com/jcabi/xml/XMLDocumentTest.java

```diff
@@ -374,4 +375,32 @@ public void preservesXmlNamespaces() throws Exception {
         );
     }
 
+    /**
+     * XMLDocument can generate a compact representation of itself.
+     * @throws Exception If something goes wrong inside
+     */
+    @Test
+    public void compacts() throws Exception {
+        final XML xml = new XMLDocument(
+            "<a> \n\t  <b/>\n <c> text </c> \n\n\t </a>"
+        );
+        final XML compact = xml.compact();
+        MatcherAssert.assertThat(
+            compact,
+            Matchers.not(
+                Matchers.equalTo(
+                    XML.class.cast(
+                        new XMLDocument("<a><b/>xyz<c> text </c></a>")
```

Comment:
You mean the string ` text ` inside of element `<c>`? Just to make it clear that the text content will keep the spaces it has (because the node contains non-whitespace characters), while the text nodes with just whitespaces such as ` \n\n\t ` will get removed. A question of clarity.